### PR TITLE
address typos, missing words and structure throughout tutorials

### PR DESCRIPTION
**Pull request by PeerHerholz**
- fix some typos
- add missing words
- make structure more homogeneous across examples
- address further questions and unclear parts in comments

First of all, I would like to apologize for any unnecessary change I made and for potentially wrong and/or misleading adaptions. I wasn't sure about one or the other thing...
I tried to limit the changes in the PR to wording, typos, missing/wrong words, punctuations, etc. . However, there were some other things I would like to address, but wasn't sure on how/where to do it best and if it's useful at all. A put together a list below and are happy to work on this further, either within this PR or a new one.

- in [2.1.2.1.](https://nistats.github.io/auto_examples/01_tutorials/plot_first_level_model_details.html#retrieving-the-data) should the citation style of the reference study be changed and a DOI link included?
- in [2.1.2.1.](https://nistats.github.io/auto_examples/01_tutorials/plot_first_level_model_details.html#retrieving-the-data) the definition of the `tr`, `events`, etc. could be explained (perhaps the TR) could be fetched through [nibabel](https://nipy.org/nibabel/)?
- in [2.1.2.2.](https://nistats.github.io/auto_examples/01_tutorials/plot_first_level_model_details.html#running-a-basic-model) the plot of the contrast could be adapted so that the respective contrasts don't overlay one another anymore
- in [2.1.2.3](https://nistats.github.io/auto_examples/01_tutorials/plot_first_level_model_details.html#changing-the-drift-model) maybe add a short explanation why the model performs poorly
- in [2.1.2.8.](https://nistats.github.io/auto_examples/01_tutorials/plot_first_level_model_details.html#masking) should the FirstLevelModel function be referenced (this accounts for all object/function mentions through out the tutorials)
- in [2.1.2.8.](https://nistats.github.io/auto_examples/01_tutorials/plot_first_level_model_details.html#masking) should code parts be highlighted (e.g. `interpolation = "nearest"`) (also throughout tutorials)
- in [2.2.1.4.](https://nistats.github.io/auto_examples/02_first_level_models/plot_adhd_dmn.html#generating-a-report) more explanations on how to view the report could be added
- in [2.2.2](https://nistats.github.io/auto_examples/02_first_level_models/plot_fir_model.html#analysis-of-an-fmri-dataset-with-a-finite-impule-response-fir-model) maybe a reference to the Neurospin Localizer could be included
- in [2.2.4.1.](https://nistats.github.io/auto_examples/02_first_level_models/plot_fiac_analysis.html#prepare-data-and-analysis-parameters) a (function) reference to `DataFrames` could be included
- in [2.2.4.3.](https://nistats.github.io/auto_examples/02_first_level_models/plot_fiac_analysis.html#generating-a-report) maybe include more information on how to view the reports (this also accounts for any other example that refers/includes a report)
- in [2.2.5.1.](https://nistats.github.io/auto_examples/02_first_level_models/plot_bids_features.html#first-level-analysis-of-a-complete-bids-dataset-from-openneuro) maybe include link to utilized dataset
- in [2.2.5.2](https://nistats.github.io/auto_examples/02_first_level_models/plot_bids_features.html#obtain-automatically-firstlevelmodel-objects-and-fit-arguments) a link/reference to `fmriprep` could be included
- in [2.2](https://nistats.github.io/auto_examples/index.html#first-level-analysis-examples) the examples [plot_fixed_effects.py](https://github.com/nistats/nistats/blob/master/examples/02_first_level_models/plot_fixed_effects.py) and [plot_predictions_residuals.py](https://github.com/nistats/nistats/blob/master/examples/02_first_level_models/plot_predictions_residuals.py) are not included in the [docs](https://nistats.github.io/auto_examples/index.html#first-level-analysis-examples)
- in [2.3.1.1.](https://nistats.github.io/auto_examples/03_second_level_models/plot_second_level_two_sample_test.html#fetch-dataset) a reference to the `localizer dataset` as utilized in the previous examples could be included
- in [2.3.2.1.](https://nistats.github.io/auto_examples/03_second_level_models/plot_thresholding.html#prepare-some-images-for-a-simple-t-test) the wording of the first sentence could be improved/adapted
- in [2.3.3.1.](https://nistats.github.io/auto_examples/03_second_level_models/plot_oasis.html#vbm-analysis-of-aging) should the list of tutorial authors be included and if so, should the way of displaying them be adapted
- in [2.3.5](https://nistats.github.io/auto_examples/03_second_level_models/plot_second_level_association_test.html#example-of-generic-design-in-second-level-models) the second sentence includes some brackets that are (at least for me) hard to grasp
- in [2.3](https://nistats.github.io/auto_examples/index.html#second-level-analysis-examples) the example [plot_proportions_activated_voxels.py](https://github.com/nistats/nistats/blob/master/examples/03_second_level_models/plot_proportion_activated_voxels.py) not in [docs](https://nistats.github.io/auto_examples/index.html#nistats-usage-examples)
- in [2.4.2](https://nistats.github.io/auto_examples/04_low_level_functions/plot_hrf.html#example-of-hemodynamic-reponse-functions) a reference to Glover could be included
- in [2.4.3](https://nistats.github.io/auto_examples/04_low_level_functions/write_events_file.html#example-of-a-events-tsv-file-generation-the-neurospin-localizer-events) the reference to Pinel could be adapted and a link added
- in [2.4.4.2.](https://nistats.github.io/auto_examples/04_low_level_functions/plot_design_matrix.html#create-design-matrices) the final plot is not showing up
- in [2.5.1.](https://nistats.github.io/auto_examples/05_complete_examples/plot_bids_analysis.html#bids-dataset-first-and-second-level-analysis) the reference to the BIDS paper could be included
- in [2.5.1.5.](https://nistats.github.io/auto_examples/05_complete_examples/plot_bids_analysis.html#second-level-model-estimation) the docs mention a "left lateralized fronto-temporal network", but the utilized plot does not allow to see that, hence it could be adapted
- [2.5.](https://nistats.github.io/auto_examples/index.html#complete-examples) the example [plot_haxby_block_classifier.py](https://github.com/nistats/nistats/blob/master/examples/05_complete_examples/plot_haxby_block_classification.py) is not included in the [docs](https://nistats.github.io/auto_examples/index.html#nistats-usage-examples)

### CODE REVIEW

**Comment by bthirion** [comment]
> in [2.1.2.1.](https://nistats.github.io/auto_examples/01_tutorials/plot_first_level_model_details.html#retrieving-the-data) should the citation style of the reference study be changed and a DOI link included?

First, of all, huge thx !
Yes.

**Comment by bthirion** [comment]
> in [2.1.2.1.](https://nistats.github.io/auto_examples/01_tutorials/plot_first_level_model_details.html#retrieving-the-data) the definition of the `tr`, `events`, etc. could be explained (perhaps the TR) could be fetched through [nibabel](https://nipy.org/nibabel/)?

Yes for explaining

I'm rather negative on fetching from nibabel. This may not be relaible (people tend to edit image headers, and software are not always consistent); For the TR at least I would rely on BIDS meta data whenever we deal with BIDS data.

**Comment by bthirion** [comment]
> * in [2.1.2.2.](https://nistats.github.io/auto_examples/01_tutorials/plot_first_level_model_details.html#running-a-basic-model) the plot of the contrast could be adapted so that the respective contrasts don't overlay one another anymore
> 
> * in [2.1.2.3](https://nistats.github.io/auto_examples/01_tutorials/plot_first_level_model_details.html#changing-the-drift-model) maybe add a short explanation why the model performs poorly
> 
> * in [2.1.2.8.](https://nistats.github.io/auto_examples/01_tutorials/plot_first_level_model_details.html#masking) should the FirstLevelModel function be referenced (this accounts for all object/function mentions through out the tutorials)
> 
> * in [2.1.2.8.](https://nistats.github.io/auto_examples/01_tutorials/plot_first_level_model_details.html#masking) should code parts be highlighted (e.g. `interpolation = "nearest"`) (also throughout tutorials)
> 
> * in [2.2.1.4.](https://nistats.github.io/auto_examples/02_first_level_models/plot_adhd_dmn.html#generating-a-report) more explanations on how to view the report could be added
> 
> * in [2.2.2](https://nistats.github.io/auto_examples/02_first_level_models/plot_fir_model.html#analysis-of-an-fmri-dataset-with-a-finite-impule-response-fir-model) maybe a reference to the Neurospin Localizer could be included

OK for all those: thx for spotting all these aspects.

**Comment by bthirion** [comment]
> in [2.2.4.1.](https://nistats.github.io/auto_examples/02_first_level_models/plot_fiac_analysis.html#prepare-data-and-analysis-parameters) a (function) reference to `DataFrames` could be included

Well, the current code is quite poor. Here, we should modify the fetcher so that it returns BIDS-compatible data. For that, we should open an issue, and maybe address it after merge into Nilearn.

**Comment by bthirion** [comment]
> * in [2.2.4.3.](https://nistats.github.io/auto_examples/02_first_level_models/plot_fiac_analysis.html#generating-a-report) maybe include more information on how to view the reports (this also accounts for any other example that refers/includes a report)
> 
> * in [2.2.5.1.](https://nistats.github.io/auto_examples/02_first_level_models/plot_bids_features.html#first-level-analysis-of-a-complete-bids-dataset-from-openneuro) maybe include link to utilized dataset
> 
> * in [2.2.5.2](https://nistats.github.io/auto_examples/02_first_level_models/plot_bids_features.html#obtain-automatically-firstlevelmodel-objects-and-fit-arguments) a link/reference to `fmriprep` could be included
> 
> * in [2.2](https://nistats.github.io/auto_examples/index.html#first-level-analysis-examples) the examples [plot_fixed_effects.py](https://github.com/nistats/nistats/blob/master/examples/02_first_level_models/plot_fixed_effects.py) and [plot_predictions_residuals.py](https://github.com/nistats/nistats/blob/master/examples/02_first_level_models/plot_predictions_residuals.py) are not included in the [docs](https://nistats.github.io/auto_examples/index.html#first-level-analysis-examples)
> 
> * in [2.3.1.1.](https://nistats.github.io/auto_examples/03_second_level_models/plot_second_level_two_sample_test.html#fetch-dataset) a reference to the `localizer dataset` as utilized in the previous examples could be included
> 
> * in [2.3.2.1.](https://nistats.github.io/auto_examples/03_second_level_models/plot_thresholding.html#prepare-some-images-for-a-simple-t-test) the wording of the first sentence could be improved/adapted

Yes for those.

**Comment by bthirion** [comment]
>     * in [2.3.3.1.](https://nistats.github.io/auto_examples/03_second_level_models/plot_oasis.html#vbm-analysis-of-aging) should the list of tutorial authors be included and if so, should the way of displaying them be adapted

Yes.

**Comment by bthirion** [comment]
> * 
> 
> * in [2.3.5](https://nistats.github.io/auto_examples/03_second_level_models/plot_second_level_association_test.html#example-of-generic-design-in-second-level-models) the second sentence includes some brackets that are (at least for me) hard to grasp
> 
> * in [2.3](https://nistats.github.io/auto_examples/index.html#second-level-analysis-examples) the example [plot_proportions_activated_voxels.py](https://github.com/nistats/nistats/blob/master/examples/03_second_level_models/plot_proportion_activated_voxels.py) not in [docs](https://nistats.github.io/auto_examples/index.html#nistats-usage-examples)
> 
> * in [2.4.2](https://nistats.github.io/auto_examples/04_low_level_functions/plot_hrf.html#example-of-hemodynamic-reponse-functions) a reference to Glover could be included
> 
> * in [2.4.3](https://nistats.github.io/auto_examples/04_low_level_functions/write_events_file.html#example-of-a-events-tsv-file-generation-the-neurospin-localizer-events) the reference to Pinel could be adapted and a link added
> 
> * in [2.4.4.2.](https://nistats.github.io/auto_examples/04_low_level_functions/plot_design_matrix.html#create-design-matrices) the final plot is not showing up
> 
> * in [2.5.1.](https://nistats.github.io/auto_examples/05_complete_examples/plot_bids_analysis.html#bids-dataset-first-and-second-level-analysis) the reference to the BIDS paper could be included
> 
> * [2.5.](https://nistats.github.io/auto_examples/index.html#complete-examples) the example [plot_haxby_block_classifier.py](https://github.com/nistats/nistats/blob/master/examples/05_complete_examples/plot_haxby_block_classification.py) is not included in the [docs](https://nistats.github.io/auto_examples/index.html#nistats-usage-examples)

OK with all your comments.

**Comment by bthirion** [comment]
> in [2.5.1.5.](https://nistats.github.io/auto_examples/05_complete_examples/plot_bids_analysis.html#second-level-model-estimation) the docs mention a "left lateralized fronto-temporal network", but the utilized plot does not allow to see that, hence it could be adapted

Indeed, the plot should show 3 views.

File: examples/02_first_level_models/plot_bids_features.py [hunk 1]
```diff
@@ -51,12 +51,12 @@
...
+# Obtain FirstLevelModel objects automatically and fit arguments
 # ---------------------------------------------------------------
-# From the dataset directory we obtain automatically FirstLevelModel objects
-# with their subject_id filled from the BIDS dataset. Moreover we obtain
-# for each model the list of run imgs and their respective events and
-# confounder regressors. Confounders are inferred from the confounds.tsv files
+# From the dataset directory we automatically obtain FirstLevelModel objects
+# with their subject_id filled from the BIDS dataset. Moreover we obtain,
+# for each model, the list of run images and their respective events and
+# confound regressors. Confounders are inferred from the confounds.tsv files
```
Comment (bthirion) [inline examples/02_first_level_models/plot_bids_features.py]:
```suggestion
# confound regressors. Those are inferred from the confounds.tsv files
```

File: examples/03_second_level_models/plot_oasis.py [hunk 2]
```diff
@@ -71,48 +71,46 @@
...
     z_map, threshold=threshold, colorbar=True, display_mode='z',
     cut_coords=[-4, 26],
     title='age effect on grey matter density (FDR = .05)')
 plotting.show()
 
 ###########################################################################
-# Can also study the effect of sex: compute the stat, compute the
-# threshold, plot the map
+# We can also study the effect of sex by computing the contrast, thresholding it
+# and plottun the resulting map.
```
Comment (bthirion) [inline examples/03_second_level_models/plot_oasis.py]:
```suggestion
# and plot the resulting map.
```

File: examples/05_complete_examples/plot_bids_analysis.py [hunk 3]
```diff
@@ -26,24 +26,24 @@
 ##############################################################################
 # Fetch example BIDS dataset
 # --------------------------
-# We download an simplified BIDS dataset made available for illustrative
+# We download a simplified BIDS dataset made available for illustrative
 # purposes. It contains only the necessary
 # information to run a statistical analysis using Nistats. The raw data
 # subject folders only contain bold.json and events.tsv files, while the
-# derivatives folder with preprocessed files contain preproc.nii and
+# derivatives folder includes thr preprocessed files preproc.nii and the
```
Comment (bthirion) [inline examples/05_complete_examples/plot_bids_analysis.py]:
```suggestion
# derivatives folder includes the preprocessed files preproc.nii and the
```

File: examples/05_complete_examples/plot_surface_bids_analysis.py [hunk 4]
```diff
@@ -124,34 +123,33 @@
 ############################################################################
 # Individual activation maps have been accumulated in the z_score_left
 # and az_scores_right lists respectively. We can now use them in a
-# group study (one -sample study)
+# group study (one -sample study).
```
Comment (bthirion) [inline examples/05_complete_examples/plot_surface_bids_analysis.py]:
```suggestion
# group study (one-sample study).
```

**Comment by PeerHerholz** [comment]
Hi @bthirion,

thank you very much for your fast reply, the feedback and all the pointers.

Also thanks for catching the mini-glitches. I actually also have a typo in the name of the branch within which I wanted to correct typos... 
As suggested by you, I work on the other points within a new branch/PR. 

Thanks again, cheers, Peer

**Comment by bthirion** [comment]
@kchawla-pi this one can merged I think.

**Pull request merged**